packages/block-editor/src/components/link-control/index.js

```diff
@@ -315,28 +495,56 @@ function LinkControl( {
 			ref={ wrapperNode }
 			className="block-editor-link-control"
 		>
-			{ isEditingLink || ! value ? (
+			{ isResolvingLink && (
+				<div className="block-editor-link-control__loading">
+					<Spinner /> { __( 'Creating' ) }…
+				</div>
+			) }
+
+			{ ( isEditingLink || ! value ) && ! isResolvingLink && (
 				<LinkControlSearchInput
 					value={ inputValue }
 					onChange={ onInputChange }
-					onSelect={ ( suggestion ) => {
-						onChange( { ...value, ...suggestion } );
-						stopEditing();
+					onSelect={ async ( suggestion ) => {
+						if (
+							suggestion.type &&
+							CREATE_TYPE === suggestion.type
+						) {
+							try {
+								cancelableOnCreate.current = makeCancelable(
+									handleOnCreate( inputValue )
+								);
+								await cancelableOnCreate.current.promise;
+								stopEditing();
+							} catch ( error ) {
+								if ( error && error.isCanceled ) {
+									return; // bail if canceled to avoid setting state
+								}
+
+								setErrorMessage(
+									__(
+										'An unknown error occurred during creation. Please try again.'
+									)
+								);
```

Comment:
Same as the `setErrorMessage` comment above in `<LinkControlSearchCreate onClick />`

Reply:
This logic has been centralised so this comment has been resolved elsewhere.